packages/core/src/components/button/buttons.tsx

```diff
@@ -73,13 +87,27 @@ export const AnchorButtonFactory = React.createFactory(AnchorButton);
 function getButtonClasses(props: IButtonProps) {
     return classNames(
         Classes.BUTTON,
-        { [Classes.DISABLED]: props.disabled },
+        { [Classes.DISABLED]: isButtonDisabled(props) },
         Classes.iconClass(props.iconName),
         Classes.intentClass(props.intent),
         props.className,
     );
 }
 
+function isButtonDisabled(props: IButtonProps) {
+    return props.disabled === true || props.loading === true;
+}
+
+function buttonContentsVisibility(loading: boolean) {
+    return {visibility: loading ? "hidden" : "visible"};
+}
+
+function maybeRenderSpinner(loading: boolean) {
+    return loading != null && loading
```

Comment:
simpy `loading` is enough, we know it's a boolean and that includes the is-defined check.